pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
+				return errors.Errorf(
+					"foreign key columns %s must be the prefix of an index", colNames(srcCols),
```

Comment:
I'm fine with not clarifying the subtlety perfectly but I think it's important that the error message make it obvious what exactly is going wrong and how to resolve the problem. I'll leave it up to you to decide whether this is sufficient - I just expect that this error message will not be particularly enlightening for users that encounter it.